src/Bubble/index.tsx

```diff
@@ -1,15 +1,19 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import classNames from 'classnames';
 
-const STYLE_VARIANTS = [
-  'primary',
-  'success',
-  'error',
-  'warning',
-];
+const STYLE_VARIANTS = ['primary', 'success', 'error', 'warning'] as const;
```

Comment:
Is it good idea to change to `enum`?
```
export enum STYLE_VARIANTS {
  PRIMARY = 'primary',
  SUCCESS = 'success',
  ...
}
```
and then in the `interface` we can
```
...
variant?: STYLE_VARIANTS,
...
```

Reply:
I agree, an enum is cleaner for the types, but `STYLE_VARIANTS` is still used in the prop types. That said, once a component has types for the props as `Bubble` now does, the `PropType` definitions become less meaningful. They'd really only used for the purpose of forming the "Props API" tables on the component pages in the documentation site, I believe.

Will we need to keep both types and prop types moving forward or I wonder if we should do some technical discovery around migrating away from prop types while keeping the same functionality with generating the props api tables.

Reply:
Yeah, I didn't use enum because it conflicts with propTypes, I didn't find a good way to reuse it in propTypes definition, so that it correctly displays in Props API table...

I think we will need to keep propTypes even when we move to TypeScript types, because they have different purpose - TypeScript types work only during compile type while propTypes during runtime, this means if we remove propTypes and consumers pass some variant that is not in `STYLE_VARIANTS` they will not get a warning that this prop is invalid. I guess they would if they have TypeScript setup inn their projects... but I believe most MFEs (if not all of them) do not use TypeScript

Reply:
> I guess they would if they have TypeScript setup inn their projects... but I believe most MFEs (if not all of them) do not use TypeScript

This is correct. The TypeScript in this PR is the first TypeScript introduced in the Open edX platform, AFAIK. It's a great callout for build time vs. runtime validation in TypeScript and propTypes; I agree it makes sense we should probably use both for the time being, until we learn otherwise.